### PR TITLE
deprecating intermediate scheduler classes

### PR DESCRIPTION
**Pull request by atravitz**
## Description
Addresses request for deprecation in issue #492 

## Motivation and Context
Improves code readability by removing unnecessary intermediate classes.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [ ] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [ ] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/environment.py
Comment (bdice) [inline flow/environment.py:397]: Our next release is 0.14, so that's when the deprecation period should begin. I think it's fine to deprecate this in one minor release, according to our deprecation policy: https://docs.signac.io/projects/flow/en/latest/support.html#deprecation-policy
```suggestion
    deprecated_in="0.14",
    removed_in="0.15",
```

File: flow/environment.py
Comment (bdice) [inline flow/environment.py:407]:
```suggestion
    deprecated_in="0.14",
    removed_in="0.15",
```

File: flow/environment.py
Comment (bdice) [inline flow/environment.py:418]:
```suggestion
    deprecated_in="0.14",
    removed_in="0.15",
```

File: flow/environment.py
Comment (bdice) [inline flow/environment.py:428]:
```suggestion
    deprecated_in="0.14",
    removed_in="0.15",
```

File: flow/environment.py
Comment (bdice) [inline flow/environment.py:430]:
```suggestion
    details="NodesEnvironment has been deprecated.",
```

**Review by bdice — APPROVED** [review]
I applied my suggested changes. @pepak13 @Charlottez112 This is ready for a second review. Please merge after reviewing if you approve.

**Review by Charlottez112 — APPROVED** [review]
Thank you for finishing this up!

**Pull request merged**
